gninasrc/lib/cnn_scorer.cpp

```diff
@@ -295,6 +295,84 @@ void CNNScorer::get_net_output(Dtype& score, Dtype& aff, Dtype& loss) {
 
   loss = lossblob->cpu_data()[0];
 }
+
+// Extract ligand atoms and coordinates
+std::pair<atomv, vecv> CNNScorer::get_ligand(const model& m) const{
+  atomv ligand_atoms = atomv(
+    m.get_movable_atoms().cbegin() + m.ligands[0].node.begin,
+    m.get_movable_atoms().cbegin() + m.m_num_movable_atoms
+  );
+
+  vecv ligand_coords = vecv(
+    m.coordinates().cbegin() + m.ligands[0].node.begin,
+    m.coordinates().cbegin() + m.m_num_movable_atoms
+  );
+
+  if(!cnnopts.flexopt){ // No flexible residues
+    CHECK_EQ(ligand_atoms.size(), m.get_movable_atoms().size());
+    CHECK_EQ(ligand_coords.size(), m.coordinates().size());
+  }
+
+  return std::pair<atomv, vecv>(ligand_atoms, ligand_coords);
+}
+
+// Concatenate movable and fixed receptor atoms
+std::tuple<atomv,vecv, std::size_t> CNNScorer::get_receptor(const model& m) const{
```

Comment:
Receptor flex and inflex atoms are separated by ligand atoms. This utility function extract flex and inflex receptor atoms and coordinates. There might be better ways of doing this. `num_flex_atoms` is also returned, needed in `CNNScorer` to extract the gradient of the movable atoms only from the receptor gradient.

Reply:
Changed to `setReceptor`